### PR TITLE
[EOC 443] Improvement refactor socket initialization and passing socket instance

### PR DESCRIPTION
**Pull request by adamklepacz**
The task is mainly related to require orders in index.js server file. As @Ciunkos said, reffering to the https://github.com/sandstreamdev/eoc/pull/289#discussion_r317961574 , import orderd should be not dependent on socketInstance initialization. 

Now it's refactored and it "works as nature always wanted 🐐 🌿 🌿".

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -28,7 +20,13 @@ const mailerRouter = require('./routes/mailer');
 const activitiesRouter = require('./routes/activity');
 const unlockLocks = require('./common/utils/unlockLocks');
 
-/* eslint-enable import/order */
+const app = express();
+const server = http.Server(app);
+
+socket.init(server);
+const socketInstance = socket.getSocketInstance(server);
+socket.listen(socketInstance);
```
Comment (alemikolo) [inline server/index.js]: proposal: could we move line 27-28 to socket `index.js` file and initialize socket listeners in `init` function? , this index app will be cleaner?

File: server/controllers/cohort.js [hunk 2]
```diff
@@ -293,7 +294,7 @@ const deleteCohortById = (req, resp) => {
       const { ownerIds: owners, name } = cohort;
       const data = { cohortId, owners };
 
-      socketActions.deleteCohort(socketInstance, allCohortsViewClients)(data);
+      socketActions.deleteCohort(io.getInstance(), allCohortsViewClients)(data);
```
Comment (Ciunkos) [inline server/controllers/cohort.js]: Please extract instance first to reduce diff and complexity.

```suggestion
      const socketInstance = io.getInstance();
      socketActions.deleteCohort(socketInstance, allCohortsViewClients)(data);
```

File: server/controllers/cohort.js [hunk 3]
```diff
@@ -350,7 +351,7 @@ const removeMember = (req, resp) => {
     })
     .then(() => {
       socketActions.removeMember(
-        socketInstance,
+        io.getInstance(),
```
Comment (Ciunkos) [inline server/controllers/cohort.js]: Please apply the same pattern everywhere.

File: server/controllers/list.js [hunk 4]
```diff
@@ -466,6 +466,7 @@ const updateListById = (req, resp) => {
     isDeleted,
     name
   });
+  const socketInstance = io.getInstance();
```
Comment (Ciunkos) [inline server/controllers/list.js]: Please move it closer to the usage.

**Pull request merged**